backend/src/main/kotlin/cz/loono/backend/api/service/PreventionService.kt

```diff
@@ -53,10 +53,12 @@ class PreventionService(
                 }
                 ?.sortedBy { it.date } ?: listOf(ExaminationRecord())
 
-            var count = 0
-            if (examsOfType != null) {
-                count = examsOfType.map { it.status == ExaminationStatusDto.CONFIRMED }.size
-            }
+            val confirmedExamsOfCurrentType = examsOfType?.filter { it.status == ExaminationStatusDto.CONFIRMED }
+            // 1) Filter all the confirmed records
+            // 2) Map all non-nullable lastExamination records
+            // 3) Find the largest or return null if the list is empty
+            val lastConfirmedDate = confirmedExamsOfCurrentType?.mapNotNull(ExaminationRecord::date)?.maxOrNull()
+            val totalCountOfConfirmedExams = confirmedExamsOfCurrentType?.size ?: 0
 
             PreventionStatusDto(
```

Comment:
I know that it is a minority - I would also rename 'date' field to something more clear like 'plannedDate' to more differentiate the dates.

Reply:
Ok, done in c8a37fa